nixos/modules/services/databases/influxdb2.nix

```diff
@@ -3,18 +3,133 @@
 let
   inherit
     (lib)
+    concatMap
+    concatMapStrings
+    count
+    elem
     escapeShellArg
+    escapeShellArgs
+    filter
+    flip
+    genAttrs
+    getExe
     hasAttr
+    hasInfix
+    head
     literalExpression
+    mkBefore
     mkEnableOption
     mkIf
     mkOption
+    optional
+    optionals
+    optionalString
     types
+    unique
     ;
 
   format = pkgs.formats.json { };
   cfg = config.services.influxdb2;
   configFile = format.generate "config.json" cfg.settings;
+
+  # A helper utility to allow provisioning tokens with deterministic secrets
+  tokenManipulator = pkgs.buildGoModule rec {
```

Comment:
This should be a separate package not to be inlined in the module.

Reply:
True, I should probably load a json file instead of baking the configuration in.